builtin/rebase.c

```diff
@@ -302,6 +303,8 @@ static int run_specific_rebase(struct rebase_options *opts)
 			opts->allow_rerere_autoupdate ?
 			"--rerere-autoupdate" : "--no-rerere-autoupdate");
 	add_var(&script_snippet, "keep_empty", opts->keep_empty ? "yes" : "");
+	add_var(&script_snippet, "gpg_sign_opt",
+		opts->gpg_sign_opt.buf ? "-S" : opts->gpg_sign_opt.buf);
```

Comment:
This is wrong. If `gpg_sign_opt` is `NULL`, it will unset the shell variable `gpt_sign_opt`, which is correct, but if it contains the string `dscho`, it won't prefix it with `-S`, it will *replace* it.

Reply:
To be clear: this `add_var()` call should be replaced by this instead:

```c
add_var(&script_snippet, "gpg_sign_opt", opts->gpg_sign_opt);
```

after changing the type of that field from `struct strbuf` to `char *`.